contracts/peripheral/ImportProxy.sol

```diff
@@ -0,0 +1,259 @@
+        (, uint256 rate,,,) = vat.ilks(WETH);
+        return muld(daiAmount, rate);
+    }
+
+    /// @dev Convert from Dai to MakerDAO debt
+    function daiToDebt(uint256 daiAmount) public view returns (uint256) {
+        (, uint256 rate,,,) = vat.ilks(WETH);
+        return divd(daiAmount, rate);
+    }
+
+    /// @dev Minimum weth needed to collateralize an amount of dai in MakerDAO
+    function wethForDai(uint256 daiAmount) public view returns (uint256) {
+        (,, uint256 spot,,) = vat.ilks(WETH);
+        return divd(daiAmount, spot);
+    }
+
+    /// @dev Minimum weth needed to collateralize an amount of fyDai in Yield. Yes, it's the same formula.
+    function wethForFYDai(uint256 fyDaiAmount) public view returns (uint256) {
+        (,, uint256 spot,,) = vat.ilks(WETH);
+        return divd(fyDaiAmount, spot);
+    }
+
+    /// @dev Amount of fyDai debt that will result from migrating Dai debt from MakerDAO to Yield
+    function fyDaiForDai(IPool pool, uint256 daiAmount) public view returns (uint256) {
+        return pool.buyDaiPreview(daiAmount.toUint128());
+    }
+
+    /// @dev Amount of dai debt that will result from migrating fyDai debt from Yield to MakerDAO
+    function daiForFYDai(IPool pool, uint256 fyDaiAmount) public view returns (uint256) {
+        return pool.buyFYDaiPreview(fyDaiAmount.toUint128());
+    }
+
+    /// @dev Internal function to transfer debt and collateral from MakerDAO to Yield
+    /// @param pool The pool to trade in (and therefore fyDai series to borrow)
+    /// @param user Vault to import.
+    /// @param wethAmount weth to move from MakerDAO to Yield. Needs to be high enough to collateralize the dai debt in Yield,
+    /// and low enough to make sure that debt left in MakerDAO is also collateralized.
+    /// @param debtAmount dai debt to move from MakerDAO to Yield. Denominated in Dai (= art * rate)
+    /// Needs vat.hope(splitter.address, { from: user });
+    /// Needs controller.addDelegate(splitter.address, { from: user });
+    function _importFromProxy(IPool pool, address user, uint256 wethAmount, uint256 debtAmount) internal {
+        IFYDai fyDai = IFYDai(pool.fyDai());
+
+        // Pool should take exactly all fyDai flash minted. ImportProxy will hold the dai temporarily
+        (, uint256 rate,,,) = vat.ilks(WETH);
+        uint256 fyDaiSold = pool.buyDai(address(this), address(this), muld(debtAmount, rate).toUint128());
+
+        daiJoin.join(address(this), dai.balanceOf(address(this)));      // Put the Dai in Maker
+        vat.frob(                           // Pay the debt and unlock collateral in Maker
+            WETH,
+            address(this),
+            address(this),
+            address(this),
+            -wethAmount.toInt256(),               // Removing Weth collateral
+            -debtAmount.toInt256()  // Removing Dai debt
+        );
+
+        // vat.flux(WETH, user, address(this), wethAmount);             // Remove the collateral from Maker
```

Comment:
Is this not needed anymore because the collateral being removed from maker is the entire collateral of the ImportProxy's vault?

Reply:
Yes, `flux` moves collateral between MakerDAO vaults, which we needed to do before calling `wethJoin.exit`. Now the collateral has been previously forked to `ImportProxy`, so no need to move it anywhere to `exit` it.